src/components/transactions.rs

```diff
@@ -146,9 +207,18 @@ pub fn create_transfer_tx(sender: Address, recipient: Address, amount: u64, wall
 }
 
 /// Convert a block template and a list of transactions into a block proposal
-pub fn template_into_proposal(block_template: BlockTemplate, mut txs: Vec<Transaction>) -> BlockProposal {
-
-    let coinbase = Transaction::read(hex::decode(block_template.coinbase_txn.data).unwrap().as_slice(), zcash_primitives::consensus::BranchId::Nu5).unwrap();
+pub fn template_into_proposal(
+    block_template: BlockTemplate,
+    mut txs: Vec<Transaction>,
+    activate: bool,
+) -> BlockProposal {
+    let coinbase = Transaction::read(
+        hex::decode(block_template.coinbase_txn.data)
+            .unwrap()
+            .as_slice(),
+        zcash_primitives::consensus::BranchId::Nu5,
```

Comment:
This code is relevant for `nu5` but probably not for `nu6`
Consider naming the function `_nu5` or make it as a parameter.

Reply:
Good point